include/fiction/algorithms/simulation/sidb/occupation_probability_of_excited_states.hpp

```diff
@@ -42,12 +42,17 @@ namespace fiction
     // Determine the minimal energy.
     const auto [energy, state_type] = *std::min_element(energy_and_state_type.cbegin(), energy_and_state_type.cend(),
                                                         [](const auto& a, const auto& b) { return a.first < b.first; });
-    const auto min_energy           = energy;
+
+    const auto min_energy = energy;
 
     // The partition function is obtained by summing up all the Boltzmann factors.
-    const double partition_function = std::accumulate(
-        energy_and_state_type.cbegin(), energy_and_state_type.cend(), 0.0, [&](const double sum, const auto& it)
-        { return sum + std::exp(-((it.first - min_energy) * 12'000 / temperature)); });
+    const double partition_function =
+        std::accumulate(energy_and_state_type.cbegin(), energy_and_state_type.cend(), 0.0,
+                        [&](const double sum, const auto& it)
+                        {
+                            return sum + std::exp(-((it.first - min_energy) * constants::physical::EV_TO_JOULE /
+                                                    (constants::physical::BOLTZMANN_CONSTANT * temperature)));
```

Comment:
The right-hand side of the addition (`std::exp(...)`) occurs three times in this file. It's worth it to factor it out as a function.

Reply:
good point!